network/topics/pubsub.go

```diff
@@ -128,7 +128,7 @@ func NewPubsub(ctx context.Context, logger *zap.Logger, cfg *PububConfig, fork f
 	var topicScoreFactory func(string) *pubsub.TopicScoreParams
 	if cfg.ScoreIndex != nil {
 		cfg.initScoring()
-		inspector := scoreInspector(logger.Named("scoreInspector"), cfg.ScoreIndex)
+		inspector := scoreInspector(logger, cfg.ScoreIndex)
```

Comment:
About Line 156:

```
	psOpts = append(psOpts, pubsub.WithEventTracer(newTracer(logger, cfg.TraceLog)))
```

We can probably not pass the pubsub option if TraceLog is false, so that libp2p will not even try to trace